### ISSUE TITLE
[java] UnnecessaryImport false positive for on-demand imports of nested classes

### ISSUE THREAD

**Issue by abyss638** [pmd/pmd#4082]
**Affects PMD Version:** 
6.48.0 (but works fine in 5.8.1)

**Rule:** 
[UnnecessaryImports](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#unnecessaryimport)

**Description:**
Nested classes (non-static and static) can be imported using on-demand import and static nested classes can be imported with on-demand static import. For both cases PMD will show 'Unused import X.X.X`

**Code Sample demonstrating the issue:**

```java
package net.sourceforge.pmd.lang.java.rule.codestyle.unnecessaryimport.package2;

public class C {
    private C() { }

    public class IC {}

    public static class ISC {}
}
```

```java
package net.sourceforge.pmd.lang.java.rule.codestyle.unnecessaryimport.package1;

import net.sourceforge.pmd.lang.java.rule.codestyle.unnecessaryimport.package2.C.*;

public class U {
    IC c;
}
```

```java
package net.sourceforge.pmd.lang.java.rule.codestyle.unnecessaryimport.package1;

import static net.sourceforge.pmd.lang.java.rule.codestyle.unnecessaryimport.package2.C.*;

public class U {
    ISC sc;
}
```

**Expected outcome:**
PMD reports a violation `Unused net.sourceforge.pmd.lang.java.rule.codestyle.unnecessaryimport.package2.C`, but that's wrong. That's a false positive.

**Running PMD through:** 
*Ant*
